theatre/core/src/projects/DefaultAssetStorage.ts

```diff
@@ -0,0 +1,172 @@
+            throw new Error(
+              `IndexedDB is required by the default asset manager, but it's not supported by this browser. To use assets, please provide your own asset manager to the project config.`,
+            )
+          },
+        }
+      }
+
+      // get all possible asset ids referenced by either static props or keyframes
+      const possibleAssetIDs = getAllPossibleAssetIDs(project)
+
+      // Clean up assets not referenced by the project. We can only do this at the start because otherwise
+      // we'd break undo/redo.
+      const idbKeys = await idb.keys<string>()
+      await Promise.all(
+        idbKeys.map(async (key) => {
+          if (!possibleAssetIDs.includes(key)) {
+            await idb.del(key)
+          }
+        }),
+      )
+
+      // Clean up idb entries exported to disk
+      await Promise.all(
+        idbKeys.map(async (key) => {
+          const assetUrl = `${baseUrl}/${key}`
+
+          const response = await fetch(assetUrl, {method: 'HEAD'})
+          if (response.ok) {
+            await idb.del(key)
+          }
+        }),
+      )
+
+      const assetsMap = new Map(await idb.entries<string, Blob>())
+      const urlCache = new Map<Blob, string>()
+
+      /** Gets idb aset url from asset blob */
+      const getUrlForAsset = (asset: Blob) => {
+        if (urlCache.has(asset)) {
+          return urlCache.get(asset)!
+        } else {
+          const url = URL.createObjectURL(asset)
+          urlCache.set(asset, url)
+          return url
+        }
+      }
+
+      /** Gets idb asset url from id */
+      const getUrlForId = (assetId: string) => {
+        const asset = assetsMap.get(assetId)
+        if (!asset) {
+          throw new Error(`Asset with id ${assetId} not found`)
+        }
+        return getUrlForAsset(asset)
+      }
+
+      return {
+        getAssetUrl: (assetId: string) => {
+          return assetsMap.has(assetId)
+            ? getUrlForId(assetId)
+            : `${baseUrl}/${assetId}`
+        },
+        createAsset: async (asset: File) => {
+          const existingIDs = getAllPossibleAssetIDs(project)
+
+          let sameSame = false
+
+          if (existingIDs.includes(asset.name)) {
+            const existingAsset =
+              assetsMap.get(asset.name) ??
+              (await fetch(`${baseUrl}/${asset.name}`).then((r) =>
+                r.ok ? r.blob() : undefined,
+              ))
+
+            if (existingAsset) {
+              // @ts-ignore
+              sameSame = await blobCompare.isEqual(asset, existingAsset)
+
+              // if same same, we do nothing
+              if (sameSame) {
+                console.log('same same', sameSame)
+                return asset.name
+              } else {
+                const renameAsset = (): boolean => {
+                  // if different, we ask the user to pls rename
+                  const newAssetName = prompt(
+                    'Please rename the asset',
```

Comment:
More descriptive prompt + suggest a new file name

Reply:
More descriptive prompt, yes, however I think it is unlikely that any suggestion would be useful. People will just end up doing more work by first deleting the suggestion and then writing their own stuff. I think prefilling the current name is the least annoying option. However if you have something specific in mind, pls let me know.